#
# The YAML file is a lightweight markup language file that supports all
# ROS parameter types. It's used by rosparam to load static parameters to
# the Parameter Server, which are commonly used to initialize part of a
# system and rarely change on run-time. Sometimes the user modifies this
# file before starting a node. This file is usually in the /config subdirectory
# of a package. Note that when dynamic reconfiguration is implemented for
# a package, most parameters will be dynamic, making this file barely used.
#
#                    _________ NOTES _________
#
# Below there's some description for the parameters and examples separated 
# in blocks.
#
# autocalibrate:      Tare and calibrate all IMUs when initializing node
# dev_names:          String identifier to find configs for each device
#                     (list dongle devices first)
# dev_type:           Device type (WRD-wired, DNG-dongle, WL-wireless)
# imu_mode:           Wireless or wired (only needed for WL devices)
# broadcast:          Switch on broadcast mode, to get data from all
#                     connected IMUs
# streaming:          Switch on streaming mode (faster)
#                     All following parameters in us:
# streaming_delay:    Sensor wait time before outputting the first packet
#                     of streaming data
# streaming_duration: Length of the streaming session ('unlimited' is 0xFFFFFFFF)
# streaming_interval: How often it should output (0 = every filter loop,
#                     other values must be >1000)
# wired_port:         Port address for device when wired
# streaming_slots:	  Defines what information each device is streaming
# wireless_dng:       Sensor to dongle association (dongle ID table must
#                     have been configured)
# wireless_id:        Logical id of WL device in dongle ID table

autocalibrate: true

## Current setup ##
dev_names: ['pc', 'pedal']
dev_type: {'pc': 'DNG', 'pedal': 'WL'}
imu_mode: {'pedal': 'wireless'}
broadcast: false
streaming: true
streaming_interval: 0
streaming_duration: 'unlimited'
streaming_delay: 0
streaming_slots: {'pedal': ['getTaredOrientationAsQuaternion']}
wired_port: {'pc': '/dev/ttyACM0'}
wireless_dng: {'pedal': 'pc'}
wireless_id: {'pedal': 7}


## 1 Wired streaming test ##
# dev_names: ['pedal']
# dev_type: {'pedal': 'WL'}
# imu_mode: {'pedal': 'wired'}
# broadcast: false
# streaming: true
# streaming_interval: 0
# streaming_duration: 'unlimited'
# streaming_delay: 0
# streaming_slots: {'pedal': ['getTaredOrientationAsQuaternion',
#                             'getNormalizedGyroRate',
#                             'getButtonState']}
# wired_port: {'pedal': '/dev/ttyACM1'}


## 2 Wired streaming test ##
# dev_names: ['pedal', 'remote']
# dev_type: {'pedal': 'WL', 'remote': 'WL'}
# imu_mode: {'pedal': 'wired', 'remote': 'wired'}
# broadcast: false
# streaming: true
# streaming_interval: 0
# streaming_duration: 'unlimited'
# streaming_delay: 0
# streaming_slots: {'pedal': ['getTaredOrientationAsQuaternion',
#                             'getNormalizedGyroRate',
#                             'getButtonState'],
#                   'remote': ['getTaredOrientationAsQuaternion',
#                              'getNormalizedGyroRate',
#                              'getButtonState']}
# wired_port: {'pedal': '/dev/ttyACM1', 'remote': '/dev/ttyACM2'}


## 1 Wireless streaming test ##
# dev_names: ['pc', 'pedal']
# dev_type: {'pc': 'DNG', 'pedal': 'WL'}
# imu_mode: {'pedal': 'wireless'}
# broadcast: false
# streaming: true
# streaming_interval: 0
# streaming_duration: 'unlimited'
# streaming_delay: 0
# streaming_slots: {'pedal': ['getTaredOrientationAsQuaternion',
#                             'getNormalizedGyroRate',
#                             'getButtonState']}
# wired_port: {'pc': '/dev/ttyACM0'}
# wireless_dng: {'pedal': 'pc'}
# wireless_id: {'pedal': 7}


## 2 Wireless streaming test ##
# dev_names: ['pc', 'pedal', 'remote']
# dev_type: {'pc': 'DNG', 'pedal': 'WL', 'remote': 'WL'}
# imu_mode: {'pedal': 'wireless', 'remote': 'wireless'}
# broadcast: false
# streaming: true
# streaming_interval: 0
# streaming_duration: 'unlimited'
# streaming_delay: 0
# streaming_slots: {'pedal': ['getTaredOrientationAsQuaternion',
#                             'getNormalizedGyroRate',
#                             'getButtonState'],
#                   'remote': ['getTaredOrientationAsQuaternion',
#                              'getNormalizedGyroRate',
#                              'getButtonState']}
# wired_port: {'pc': '/dev/ttyACM0'}
# wireless_dng: {'pedal': 'pc', 'remote': 'pc'}
# wireless_id: {'pedal': 1, 'remote': 7}


## 3 Wireless streaming test ##
# dev_names: ['pc', 'pedal', 'remote1', 'remote2']
# dev_type: {'pc': 'DNG', 'pedal': 'WL', 'remote1': 'WL', 'remote2': 'WL'}
# imu_mode: {'pedal': 'wireless', 'remote1': 'wireless', 'remote2': 'wireless'}
# broadcast: false
# streaming: true
# streaming_interval: 0
# streaming_duration: 'unlimited'
# streaming_delay: 0
# streaming_slots: {'pedal': ['getTaredOrientationAsQuaternion',
#                             'getNormalizedGyroRate',
#                             'getButtonState'],
#                   'remote1': ['getTaredOrientationAsQuaternion',
#                               'getNormalizedGyroRate',
#                               'getButtonState'],
#                   'remote2': ['getTaredOrientationAsQuaternion',
#                               'getNormalizedGyroRate',
#                               'getButtonState']}
# wired_port: {'pc': '/dev/ttyACM2'}
# wireless_dng: {'pedal': 'pc', 'remote1': 'pc', 'remote2': 'pc'}
# wireless_id: {'pedal': 4, 'remote1': 5, 'remote2': 1}


## 4 Wireless streaming test ##
# dev_names: ['pc', '1', '3', '7', '8']
# dev_type: {'pc': 'DNG', '1': 'WL', '3': 'WL', '7': 'WL', '8': 'WL'}
# imu_mode: {'1': 'wireless', '3': 'wireless', '7': 'wireless', '8': 'wireless'}
# broadcast: false
# streaming: true
# streaming_interval: 0
# streaming_duration: 'unlimited'
# streaming_delay: 0
# streaming_slots: {'1': ['getTaredOrientationAsQuaternion'],
#                   '3': ['getTaredOrientationAsQuaternion'],
#                   '7': ['getTaredOrientationAsQuaternion'],
#                   '8': ['getTaredOrientationAsQuaternion']}
# wired_port: {'pc': '/dev/ttyACM0'}
# wireless_dng: {'1': 'pc', '3': 'pc', '7': 'pc', '8': 'pc'}
# wireless_id: {'1': 1, '3': 3, '7': 7, '8': 5}
